### DISCUSSION TITLE
Black Hole Improvements

### DISCUSSION THREAD

**Discussion by TemperedFool** [InnovativeOnlineIndustries/Industrial-Foregoing#1007] [Feature Requests]
I have a few suggestions for improving Black Hole Units and Tanks:

~~- Voiding of some form: I think the power of Voiding upgrade drawers is too high, but the ability to void excess (even with max-int storage) can be very useful in not destroying a server, and simplifying setups. I think I would suggest some way to add and filter a trash can/void upgrade to the BH Controller, rather than allowing any given Unit/Tank to void.~~ I stand corrected, did not realize this was already there. 
- Redstone emission based on fill level. I would love to see this take a two different forms:
- Emission proportional to fill level; would be good if this can be switched 0-15 and 15-0
- Emission when the fill level hits a specific configurable point. Would be good if we can choose between <= and >=. Bonus points if we can set what level of redstone is emitted.
- For the controller, I'd like to see some ability to combine the outputs of the various BH's inside of it - emit lowest signal, emit highest signal, and combine/add signals come to mind.

I'm personally thinking these are some sort of upgrade to the existing BH's, but being built in would be fine. For my purposes, the RS Signalling is more critical on the lower tier containers, as they are more likely to actually fill up.

**Comment by CXWRTH** [comment]
I'm playing through ATM6 with industrial-foregoing-1.16.5-3.2.10.1-decb7ae

I **love** that Black Hole Tanks/Units now have a progression system.

I **do not love** that they don't have upgrade paths. It's pretty frustrating having old units/tanks lying around and having to arrange liquid/item transfers to the new tanks. Also, in some cases the pipes configured in machine setups break their configs when the block they are connected to is removed so upgrading some of my machines is now also requiring reconfiguring the machine so it works again.
